managedtenants/bundles/binary_deps.py

```diff
@@ -1,5 +1,25 @@
 from sretoolbox.binaries import Mtcli, OperatorSDK, Opm
 
-OPM = Opm(version="1.19.5", download_path="/tmp")
-MTCLI = Mtcli(version="0.10.0", download_path="/tmp")
-OPERATOR_SDK = OperatorSDK(version="1.4.2", download_path="/tmp")
+
+class LazyBin:
+    """
+    Abstraction around sretoolbox to avoid pulling binaries from the internet
+    before they actually need to be ran.
+    """
+
+    def __init__(self, bin_class, version, download_path):
+        self.bin_class = bin_class
+        self.version = version
+        self.download_path = download_path
+        self.instance = None
+
+    def run(self, cmd):
+        if self.instance is None:
+            self.instance = self.bin_class(self.version, self.download_path)
+
+        self.instance.run(*cmd)
+
+
+OPM = LazyBin(Opm, version="1.19.5", download_path="/tmp")
+MTCLI = LazyBin(Mtcli, version="0.10.0", download_path="/tmp")
+OPERATOR_SDK = LazyBin(OperatorSDK, version="1.4.2", download_path="/tmp")
```

Comment:
can we extract out these versions from here?

Reply:
other than that looks good

Reply:
I am sorry but I am not sure I understand. What do you mean?

Reply:
can we not hard-code these version here? extract out in a config file?

Reply:
I think extracting those versions to a config file, and giving a user the ability to change them at runtime is dangerous. The `managedtenants/bundles` package depends on these versions, and is only tested for these versions. I think we need to keep them hardcoded, and test any change with the whole PR->review->release process.